diff --git "a/Dmf/Modules.Library/Dmf_VirtualHidMini.c" "b/Dmf/Modules.Library/Dmf_VirtualHidMini.c"
index 960fe58..79d64c6 100644
--- "a/Dmf/Modules.Library/Dmf_VirtualHidMini.c"
+++ "b/Dmf/Modules.Library/Dmf_VirtualHidMini.c"
@@ -340,16 +340,23 @@ Return Value:
     // Get report Id from output buffer length.
     //
     ntStatus = WdfRequestRetrieveOutputMemory(Request,
-                                              &outputMemory);
-    if ( !NT_SUCCESS(ntStatus) )
+        &outputMemory);
+    if ( !NT_SUCCESS(ntStatus) && ntStatus != STATUS_BUFFER_TOO_SMALL )
     {
         TraceEvents(TRACE_LEVEL_ERROR, DMF_TRACE, "WdfRequestRetrieveOutputMemory fails: ntStatus=%!STATUS!", ntStatus);
         goto Exit;
     }
 
-    WdfMemoryGetBuffer(outputMemory,
-                       &outputBufferLength);
-    Packet->reportId = (UCHAR) outputBufferLength;
+    if ( ntStatus == STATUS_BUFFER_TOO_SMALL )
+    {
+        Packet->reportId = 0;
+    }
+    else
+    {
+        WdfMemoryGetBuffer(outputMemory,
+            &outputBufferLength);
+        Packet->reportId = (UCHAR)outputBufferLength;
+    }
 
     // Get report buffer from input buffer.
     //
